src/example.py

```diff
@@ -0,0 +1,216 @@
+import jax
+import jax.numpy as jnp
+import numpy as np
+
+from collections import namedtuple
+import argparse
+
+import os
+
+import math
+import time
+
+from data import import_data_efficient_mask
+from dp_accounting_utils import compute_epsilon, calculate_noise
+
+from jax_mask_efficient import (
+    create_train_state,
+    compute_per_example_gradients,
+    add_trees,
+    process_a_physical_batch,
+    model_evaluation,
+    noise_addition,
+    update_model,
+)
+
+os.environ["XLA_PYTHON_CLIENT_PREALLOCATE"] = "false"
+os.environ["XLA_PYTHON_CLIENT_MEM_FRACTION"] = ".90"
+os.environ["XLA_PYTHON_CLIENT_ALLOCATOR"] = "platform"
+
+def _parse_arguments(args, dataset_size):
+    num_steps = args.epochs * math.ceil(dataset_size / args.bs)
+
+    q = 1 / math.ceil(dataset_size / args.bs)
+
+    noise_std = calculate_noise(q, args.epsilon, args.target_delta, args.epochs, args.accountant)
+    C = args.grad_norm
+
+    optimizer_config = namedtuple("Config", ["momentum", "learning_rate"])
```

Comment:
Since we are using `optax.adam`, the momentum parameter is not required. Furthermore, it's not passed to the optimizer later on in the `example.py`.

Reply:
I removed it now, only leaving the learning_rate. Is passing this named_tuple kinda the default in JAX?

Reply:
I don't think it is a default. It's just for convenience of separating the optimizer parameters into a separate object. You might as well pass a dictionary, and change how the values are accessed.